services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -3005,6 +2971,172 @@ private Pair<Boolean, Version> addVersion(
     }
   }
 
+  /**
+   * Determines whether real-time topics should be created for the given store and version.
+   *
+   * <p>Real-time topics are created based on the following conditions:
+   * <ul>
+   *   <li>The store and version must both be hybrid ({@code store.isHybrid()} and {@code version.isHybrid()}).</li>
+   *   <li>If the controller is a child, real-time topics are always created.</li>
+   *   <li>If the controller is a parent, real-time topics are created only if:
+   *     <ul>
+   *       <li>Active-active replication is disabled for the store, and</li>
+   *       <li>Either the store's data replication policy is {@code DataReplicationPolicy.AGGREGATE}, or</li>
+   *       <li>Incremental push is enabled for the store.</li>
+   *     </ul>
+   *   </li>
+   * </ul>
+   *
+   * @param store the store being evaluated
+   * @param version the version being evaluated
+   * @return {@code true} if real-time topics should be created; {@code false} otherwise
+   */
+  boolean isRealTimeTopicRequired(Store store, Version version) {
+    if (!store.isHybrid() || !version.isHybrid()) {
```

Comment:
no problem with this, but can the newly created `version`'s "hybridness" be different from that of its `store` ?